SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -102,13 +126,15 @@ - (void)sdl_accessoryDisconnected:(NSNotification *)notification {
     EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
     if (accessory.connectionID == self.session.accessory.connectionID) {
         self.sessionSetupInProgress = NO;
+        [self sdl_backgroundTaskEnd];
         [self disconnect];
         [self.delegate onTransportDisconnected];
     }
 }
 
 - (void)sdl_applicationWillEnterForeground:(NSNotification *)notification {
     [SDLDebugTool logInfo:@"App Foregrounded Event" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    [self sdl_backgroundTaskEnd];
```

Comment:
We do need to explicitly end background tasks when we are done with background processing. The expiration handler is only our last chance to end the background task before the OS force terminates the app. I've thought carefully about the cases where the task should be cleaned up. It's possible I'm missing something, so I'd love feedback on why you believe we don't need the call to occur when switching to the foreground.